src/main/java/org/jabref/gui/sidepane/SidePaneContainerView.java

```diff
@@ -0,0 +1,176 @@
+        if (viewModel.show(sidePane)) {
+            updateView();
+            if (sidePane == GROUPS) {
+                ((GroupsSidePaneView) getSidePaneView(sidePane)).afterOpening();
+            }
+        }
+    }
+
+    private void hide(SidePaneType sidePane) {
+        if (viewModel.hide(sidePane)) {
+            updateView();
+        }
+    }
+
+    private void moveUp(SidePaneType sidePane) {
+        if (viewModel.moveUp(sidePane)) {
+            updateView();
+        }
+    }
+
+    private void moveDown(SidePaneType sidePane) {
+        if (viewModel.moveDown(sidePane)) {
+            updateView();
+        }
+    }
+
+    /**
+     * If the given component is visible it will be hidden and the other way around.
+     */
+    private void toggle(SidePaneType sidePane) {
+        if (viewModel.isSidePaneVisible(sidePane)) {
+            hide(sidePane);
+        } else {
+            show(sidePane);
+        }
+    }
+
+    private void updateView() {
+        showVisibleSidePanes();
+        setVisible(!viewModel.getVisiblePanes().isEmpty());
+    }
+
+    public BooleanProperty sidePaneVisibleProperty(SidePaneType sidePane) {
+        return switch (sidePane) {
+            case GROUPS -> viewModel.groupsSidePaneVisibleProperty();
+            case WEB_SEARCH -> viewModel.webSearchSidePaneVisibleProperty();
+            case OPEN_OFFICE -> viewModel.openOfficePaneVisibleProperty();
+        };
+    }
+
+    public ToggleCommand getToggleCommandFor(SidePaneType sidePane) {
+        return new ToggleCommand(sidePane);
+    }
+
+    private class CloseSidePaneAction extends SimpleCommand {
+        private final SidePaneType toCloseSidePane;
+
+        public CloseSidePaneAction(SidePaneType toCloseSidePane) {
+            this.toCloseSidePane = toCloseSidePane;
+        }
+
+        @Override
+        public void execute() {
+            hide(toCloseSidePane);
+        }
+    }
+
+    private class MoveUpAction extends SimpleCommand {
+        private final SidePaneType toMoveUpSidePane;
+
+        public MoveUpAction(SidePaneType toMoveUpSidePane) {
+            this.toMoveUpSidePane = toMoveUpSidePane;
+        }
+
+        @Override
+        public void execute() {
+            moveUp(toMoveUpSidePane);
+        }
+    }
+
+    private class MoveDownAction extends SimpleCommand {
+        private final SidePaneType toMoveDownSidePane;
+
+        public MoveDownAction(SidePaneType toMoveDownSidePane) {
+            this.toMoveDownSidePane = toMoveDownSidePane;
+        }
+
+        @Override
+        public void execute() {
+            moveDown(toMoveDownSidePane);
+        }
+    }
+
+    public class ToggleCommand extends SimpleCommand {
+
+        private final SidePaneType sidePane;
+
+        public ToggleCommand(SidePaneType sidePane) {
+            this.sidePane = sidePane;
+        }
+
+        @Override
+        public void execute() {
+            toggle(sidePane);
+        }
+    }
+}
```

Comment:
Do we need to integrate this in our Command infrastructure, or would be supplies enough?
I don't think we need to track these commands in telemetry or undo/redo that.. @tobiasdiez

Reply:
Right now it produces a lot of boilerplate.

Reply:
I would keep it like this, inheriting from `SimpleCommand`. Otherwise, it is harder to use them in menus.

Reply:
But this does only apply to toggleaction.